xio-core/src/main/java/com/xjeffrose/xio/http/ClientChannelResponse.java

```diff
@@ -0,0 +1,20 @@
+package com.xjeffrose.xio.http;
+
+import javax.annotation.Nullable;
+import lombok.Getter;
+import lombok.experimental.Accessors;
+
+public class ClientChannelResponse {
+  @Accessors(fluent = true)
+  @Getter
+  private boolean isDone;
+
+  @Getter private boolean isSuccess;
+  @Getter private Throwable cause;
+
+  public ClientChannelResponse(boolean isDone, boolean isSuccess, @Nullable Throwable cause) {
+    this.isDone = isDone;
+    this.isSuccess = isSuccess;
+    this.cause = cause;
+  }
+}
```

Comment:
If this only exists while the class is under test, we can make it an inner class of the test itself. If we need it in another class we can put it in `src/test/java/com/xjeffrose/xio/http`

Reply:
ok i can do that =p

Reply:
ok after discussing with pdex, this can stay